SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/RPCTypes.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Interop/RPCs/RPCTypes.h"
+
+namespace SpatialGDK
+{
+RPCWritingContext::EntityWrite::EntityWrite(EntityWrite&& Write)
+	: EntityId(Write.EntityId)
+	, ComponentId(Write.ComponentId)
+	, Ctx(Write.Ctx)
+	, Fields(Write.Fields)
+{
+	Data = Write.Data;
+	Write.bActiveWriter = false;
+	Write.Fields = nullptr;
+	Write.Data = nullptr;
+}
+
+RPCWritingContext::EntityWrite::~EntityWrite()
+{
+	if (bActiveWriter)
+	{
+		switch (Ctx.Kind)
+		{
+		case DataKind::Generic:
+			break;
+		case DataKind::ComponentData:
+			if (Ctx.DataWrittenCallback)
```

Comment:
Since DataKind and the type of callback seem to be strongly coupled can the callback be part of the RPCWritingContext constructor? That way the callback members can be private and we can't assign the incorrect callback for a given type.

Reply:
Make sense.